scipy/stats/_qmc.py

```diff
@@ -1335,19 +1339,30 @@ def __init__(
             )
 
         self.bits = bits
-        if self.bits > 64:
+        self.dtype_i: type
+        self.dtype_f: type
+
+        if self.bits is None:
+            self.bits = 30
+            self.dtype_f = np.float64
+
+        if self.bits <= 32:
```

Comment:
We could do this. But the resolution might be very low. Between 0-1, it does not leave much possibilities.

```python
2**16
# 65536
np.finfo(np.float16)
# finfo(resolution=0.001, min=-6.55040e+04, max=6.55040e+04, dtype=float16)
```

It's more or less the same problem with 32 bits but the limit is higher and will saturate further down the line which makes it less of an issue I think.

```python
2**32
# 4294967296
np.finfo(np.float32)
# finfo(resolution=1e-06, min=-3.4028235e+38, max=3.4028235e+38, dtype=float32)
```

But maybe the output type should be different and not linked to `bits`. I linked both for simplicity for the user.

Reply:
I am actually asking myself if we should not be able to store at least of the possibilities. Like for 32 bits, we should have float64 because there is just a resolution of 1e-6 which is not enough (if my interpretation is correct). I will make some tests and report back. This might simplify the deprecation decision 👍

Reply:
> We could do this. But the resolution might be very low.

Ah, makes sense. Yeah, let's avoid float16 then!

Reply:
> Like for 32 bits, we should have float64 because there is just a resolution of 1e-6 which is not enough (if my interpretation is correct). I will make some tests and report back. This might simplify the deprecation decision

Sounds good. I see returning only float64 has two benefits:

- Simple for the user and doesn't break the backward compatibility.
- Higher precision

So, I think just returning float64 would be better than to switch to float32 for lower bits.

Reply:
Ok so I did some testing. With `bits=32`, if we sample 2^25 then some elements appears twice. Increasing the precision to float64 removes this issue. So for now I will simplify things and just use float64 for the return type.

We could add `dtype` parameter but it would mean that the user would have to know the limits of the type they use with respect to the number of bits they use. I am not sure how to calculate this exactly. It might be confusing anyway.

I will remove the fused type on the return in the Cython part.